components/datetime/Cargo.toml

```diff
@@ -37,6 +37,7 @@ tinystr = { version = "0.4.5" }
 serde = { version = "1.0", features = ["derive"], optional = true }
 smallvec = "1.6"
 thiserror = "1.0"
+yoke = { version = "0.2", path = "../../utils/yoke", features = ["serde", "derive"] }
```

Comment:
Issue: I don't like adding direct `yoke` dependencies everywhere, since they are largely redundant with `icu_provider`.  Suggestion: use my re-exported `icu_provider::yoke` instead, or define macros in `icu_provider` that delegate to the `yoke` ones.

More specifically, I envision `#[derive(DataStruct)]` which expands to `#[derive(Yokeable, ZeroCopyFrom, DataMarker)]`.

Reply:
I can't use `icu_provider::yoke`, the paths are generated in the custom derive. I'm not going to make the custom derive assume the existence of `icu_provider::yoke`, that's too specialized.

Once we do have `derive(DataStruct)` we can paper over this.

Reply:
Actually I have a way to handle this.

Reply:
I handled it. It's not _great_, but we'll be papering over it anyway. Ultimately the proc macro has to assume that the crate `yoke` is in scope with that name _somehow_, and normal Rust style is to always use the crate in the Cargo.toml (as opposed to getting it via something else) so that you don't have to ensure the name exists locally.

Reply:
There may be a better way to do this

Reply:
Nope. This is the best I can do

Reply:
Okay, thanks, this looks good and we can improve it with `#[derive(DataStruct)]`